pkgs/build-support/setup-hooks/auto-patchelf.sh

```diff
@@ -115,17 +127,17 @@ autoPatchelfFile() {
 
     # We're going to find all dependencies based on ldd output, so we need to
     # clear the RPATH first.
-    patchelf --remove-rpath "$toPatch"
+    runPatchelf --remove-rpath "$toPatch"
 
-    local missing="$(
+    local missing
+    missing="$(
```

Comment:
This actually breaks autoPatchelf. 

See https://tldp.org/LDP/abs/html/localvar.html what Evgeniy Ivanov points out.
When ldd complains that file is not a dynamic executable, it return exit code 1. By this refactoring this exit code bubbles up and builder fails. It took me some time to figure this out.

Reply:
I'm sorry for your loss of time through this. I will take more care next time when applying shellcheck recommendations.
The problem is already solved via https://github.com/NixOS/nixpkgs/pull/106830

Hopefully such incidents will become less likely as soon as shellcheck is going to be incorporated as a github action.